test/test_resize.cpp

```diff
@@ -2758,6 +2758,87 @@ TEST_F(ResizeTest, SliceAndReshapeRepro540Manual) {
   }
 }
 
+// Test concretizing a pad that follows a reshape. This requires the
+// ExpressionEvaluator used in concretization to propagate shapes properly
+// across symbolic reshapes in order to infer the size of the downstream pad.
+TEST_F(ResizeTest, ReshapeToPad) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  TensorView* tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+
+  auto s0 = IrBuilder::create<Val>(DataType::Int);
+  auto s1 = IrBuilder::create<Val>(DataType::Int);
+  auto s2 = IrBuilder::create<Val>(DataType::Int);
+  auto s3 = IrBuilder::create<Val>(DataType::Int);
+  fusion.addInput(s0);
+  fusion.addInput(s1);
+  fusion.addInput(s2);
+  fusion.addInput(s3);
+
+  auto tv1 = reshape(tv0, {s2, s3});
+  auto tv2 = pad(tv1, {fusion.zeroVal(), s0, fusion.zeroVal(), s1});
+  fusion.addOutput(tv2);
+
+  FusionExecutorCache fusion_executor_cache(std::move(fusion_ptr));
```

Comment:
Can you please assert that the fusion is indeed segmented? That's not ideal and we definitely want to improve that.